tests/test_external_search.py

```diff
@@ -1018,283 +838,978 @@ def expect_ids(works, *query_args):
+        # certain string left over.
+        def assert_parses_as(query_string, *matches):
+            matches = list(matches)
+            parser = QueryParser(query_string, MockQuery)
+            remainder = matches.pop(-1)
+            if remainder:
+                remainder_match = MockQuery.simple_query_string_query(
+                    remainder, query_string_fields
+                )
+                matches.append(remainder_match)
+            eq_(matches, parser.match_queries)
+            eq_(query_string, parser.original_query_string)
+            eq_(remainder, parser.final_query_string)
+
+        # Here's the same test from before, using the new
+        # helper function.
+        assert_parses_as(
+            "science fiction about dogs",
+            ("genres.name", "Science Fiction"),
+            "about dogs"
+        )
+
+        # Test audiences.
+
+        assert_parses_as(
+            "children's picture books",
+            ("audience", "Children"),
+            "picture books"
+        )
+
+        # (It's possible for the entire query string to be eaten up,
+        # such that there is no remainder match at all.)
+        assert_parses_as(
+            "young adult romance",
+            ("genres.name", "Romance"),
+            ("audience", "YoungAdult"),
+            ''
+        )
+
+        # Test fiction/nonfiction status.
+        assert_parses_as(
+            "fiction dinosaurs",
+            ("fiction", "Fiction"),
+            "dinosaurs"
+        )
+
+        # (Genres are parsed before fiction/nonfiction; otherwise
+        # "science fiction" would be chomped by a search for "fiction"
+        # and "nonfiction" would not be picked up.)
+        assert_parses_as(
+            "science fiction or nonfiction dinosaurs",
+            ("genres.name", "Science Fiction"), ("fiction", "Nonfiction"),
+            "or  dinosaurs"
+        )
 
-    def test_query_works_from_lane_definition_handles_medium(self):
-        search = DummyExternalSearchIndex()
+        # Test target age.
 
-        lane = self._lane("Only Audio")
-        lane.media = [Edition.AUDIO_MEDIUM]
-        filter = search.make_filter(
-            [self._default_collection.id],
-            lane.media, lane.languages,
-            lane.fiction, lane.audiences, lane.target_age,
-            lane.genre_ids, lane.customlist_ids,
+        assert_parses_as(
+            "grade 5 science",
+            ("genres.name", "Science"), ((10, 10), 40),
+            ''
         )
-        collection_filter, medium_filter = filter['and']
-        expect = dict(terms=dict(medium=[Edition.AUDIO_MEDIUM.lower()]))
-        eq_(expect, medium_filter)
 
-    def test_query_works_from_lane_definition_handles_age_range(self):
-        search = DummyExternalSearchIndex()
+        assert_parses_as(
+            'divorce ages 10 and up',
+            ((10, 14), 40),
+            'divorce  and up' # TODO: not ideal
```

Comment:
yeah that's not great - did this work before?

Reply:
also how did it get 14?

Reply:
14 is YOUNG_ADULT_AGE_CUTOFF, so default_audience_for_target_range interprets "10 and up" as "10-14".

I don't think the "and up" part was chopped out before.